client/state/plugins/installed/selectors.js

```diff
@@ -0,0 +1,164 @@
+import filter from 'lodash/filter';
+import forEach from 'lodash/forEach';
+import values from 'lodash/values';
+import sortBy from 'lodash/sortBy';
+import find from 'lodash/find';
+import uniq from 'lodash/uniq';
+import compact from 'lodash/compact';
+import omit from 'lodash/omit';
+import every from 'lodash/every';
+import some from 'lodash/some';
+
+const _filters = {
+	none: function() {
+		return false;
+	},
+	all: function() {
+		return true;
+	},
+	active: function( plugin ) {
+		return plugin.sites.some( function( site ) {
+			return site.plugin && site.plugin.active;
+		} );
+	},
+	inactive: function( plugin ) {
+		return plugin.sites.some( function( site ) {
+			return site.plugin && ! site.plugin.active;
+		} );
+	},
+	updates: function( plugin ) {
+		return plugin.sites.some( function( site ) {
+			return site.plugin && site.plugin.update && site.canUpdateFiles;
+		} );
+	},
+	isEqual: function( pluginSlug, plugin ) {
+		return plugin.slug === pluginSlug;
+	}
+};
+
+const isRequesting = function( state, siteId ) {
+	// if the `isRequesting` attribute doesn't exist yet,
+	// we assume we are still launching the fetch action, so it's true
+	if ( typeof state.plugins.installed.isRequesting[ siteId ] === 'undefined' ) {
+		return true;
+	}
+	return state.plugins.installed.isRequesting[ siteId ];
+};
+
+const isRequestingForSites = function( state, sites ) {
+	// As long as any sites have isRequesting true, we consider this group requesting
+	return some( sites, ( siteId ) => isRequesting( state, siteId ) );
+};
+
+const hasRequested = function( state, siteId ) {
+	if ( typeof state.plugins.installed.hasRequested[ siteId ] === 'undefined' ) {
+		return false;
+	}
+	return state.plugins.installed.hasRequested[ siteId ];
+};
+
+const hasRequestedForSites = function( state, sites ) {
+	// All sites are requested when they all return hasRequested true
+	return every( sites, ( siteId ) => hasRequested( state, siteId ) );
+};
+
+const getPlugins = function( state, sites, pluginFilter = false ) {
+	let pluginList = {};
+	forEach( sites, ( site ) => {
+		const list = state.plugins.installed.plugins[ site.ID ] || [];
+		list.map( ( item ) => {
+			// plugin has a sites property which lists the sites this plugin is installed on
+			// each site, in turn, has a plugin property for the specific plugin's info.
+			const siteWithPlugin = Object.assign( {}, site, { plugin: item } );
+			if ( pluginList[ item.slug ] ) {
+				pluginList[ item.slug ].sites.push( siteWithPlugin );
+			} else {
+				pluginList[ item.slug ] = Object.assign( {}, item, { sites: [ siteWithPlugin ] } );
+			}
+		} );
+	} );
+	if ( !! pluginFilter && _filters[ pluginFilter ] ) {
```

Comment:
I don't see any need to coerce to a boolean here. `if ( pluginFilter &&` should work just as well